isisd/isis_tlvs.c

```diff
@@ -3042,7 +3069,8 @@ static int unpack_tlv_router_cap(enum isis_tlv_context context,
 			/* Only one range is supported. Skip subsequent one */
 			size = length - (size + SUBTLV_SR_BLOCK_SIZE);
 			if (size > 0)
-				stream_forward_getp(s, length);
+				stream_forward_getp(s, size);
+
 			break;
 		case ISIS_SUBTLV_ALGORITHM:
 			/* Only 2 algorithms are supported: SPF & Strict SPF */
```

Comment:
We may miss a check on `length` before calling `stream_get`. We may need a check `length != 0` here.

Reply:
Fixed.